{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%matplotlib inline\n",
    "import matplotlib\n",
    "import seaborn as sns\n",
    "matplotlib.rcParams['savefig.dpi'] = 144"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "# Identifying Driver Mutations that Predict Prognosis in Squamous Cell Carcinomas Using TCGA Data\n",
    "\n",
    "## Data Incubator Capstone, Summer 2017, Eric Jaehnig\n",
    "\n",
    "### Introduction\n",
    "\n",
    "The Cancer Genome Atlas (TCGA) contains data from genomic (i.e., mutations, copy number aberrations), transcriptomic (gene expression), and proteomic (protein levels) analysis of human cancer samples from 150 studies (see http://www.cbioportal.org/faq.jsp). Associated with these large biological datasets are various types of clinical data describing characteristics of the patients from which the samples were obtained. For this project, I propose to use mutation data, the most abundant data type in TCGA, to identify potential cancer driver mutations for squamous cell carcinomas. Of the 150 datasets deposited in TCGA, 11 include mutations identified from whole genome or whole exome sequencing of 5 different types of squamous cell cell carcinomas from 2200 cancer patients. My goal is to determine if there is a set of common genes that is mutated in squamous cell cancers and to identify the cellular processes and pathways that these genes regulate. Since these genes likely drive carcinogenesis in squamous epithelial cells, drugs that target the pathways they are involved in could potentially be used to treat squamous cell cancers in the future. While a quick Pubmed search reveals that specific squamous cell carcinoma or broad pan-cancer analysis of TCGA data is common, I only found one study focused on one gene that involved pan-cancer analysis specifically for squamous cell cancers, lending credence to the novelty of the proposed project.\n",
    "\n",
    "#### Data sources and preliminary analysis\n",
    "\n",
    "I downloaded the datasets for each of the 5 squamous cell carcinomas: cervical, esophageal, head and neck, lung, and skin. I loaded the “data_mutations_extended.txt” files for each of the 11 datasets into R dataframes. Each row in these files contains a uniquely identified mutation (multiple mutations per patient, each of which occupies a separate row). I selected only columns that may be of potential use when populating the dataframes, including gene names, the specific nucleotide changes, the corresponding amino acid changes, and the impact that these changes are predicted to have on the gene’s function. Since some mutations are predicted to have little or no effect on function, I focused on genes for which the predicted impact is “medium” or “high”. I then determined the set of genes that were mutated for each of the 5 squamous cell cancers. \n",
    "\n",
    "Since cancer is marked by genomic instability, which can lead to passenger mutations in several genes that don’t contribute carcinogenesis, I decided to focus on genes that were more likely to be driver mutations that promote cancer progression. To do this, I generated the Venn diagram shown in Plot 1. This diagram shows how the 5 sets of genes overlap. My rationale for focusing on the set of genes that was mutated in all of the squamous cell carcinomas has two components. The first is that passenger mutations are more likely to be random, while driver mutations are more likely to occur more consistently; thus, focusing on genes mutated in all squamous cell carcinomas is a strategy for weeding out passenger mutations. The second is that my goal is to identify the mutated genes that drive squamous cell carcinogenesis in general, not specific types of squamous cell carcinomas. I found that there were approximately 14,000 unique genes mutated in lung and head and neck cancer, ~12,000 in cervical cancer, ~4500 in esophageal cancer, and ~500 mutated in skin cancer. Of these genes, only 193 were mutated in all 5 cancers (center of Plot 1, download pdf here: https://agile-ocean-41073.herokuapp.com/).\n",
    "\n",
    "Next, I evaluated this set of genes for cellular pathway (KEGG) enrichment because mutation of multiple genes in the same pathway suggests that the pathway plays an important role in carcinogenesis. 89 KEGG pathways showed statistically significant enrichment for genes mutated in all of the squamous cell cancers. I chose to analyze one of these, the second most significant pathway, more closely by plotting the mutated genes to pathway in Plot 2 (https://limitless-stream-36820.herokuapp.com/). The plot shows all of the genes in the pathway, EGFR tyrosine kinase inhibitor resistance, with the genes mutated in squamous cell cancers colored red. While most of these genes already have established roles in cancer, the fact that squamous cell cancers show such strong enrichment for mutations in genes involved in resistance to EGFR inhibitors suggests that these drugs may have limited efficacy in treating squamous cell carcinomas in the long-term. \n",
    "\n",
    "#### Proposed strategy for data analysis\n",
    "\n",
    "More informative analysis of these genes is possible when considering the characteristics of the patient population, which is what I propose to do for the fellowship. Each dataset also contains a “data_clinical.txt” file that includes each patient’s age, sex, and disease status. Since the disease status field indicates whether the patient was disease-free or had cancer that either recurred or progressed when the studies were undertaken, machine learning approaches will be used to try classifying the set of mutated genes into genes associated with a good prognosis and into genes associated with poor prognosis. Genes that are associated with good prognosis may serve as biomarkers that could be used to predict response to treatment, while drugs that target genes associated with poor prognosis may provide improved outcomes for these patients. \n",
    "\n",
    "Finally, I will create a web-based app that will allow researchers to input their gene of interest and display the location of that gene in the Venn diagram above (Plot 1). The region of the Venn diagram in which the gene falls will be color coded to indicate that the gene was found to be mutated in the SCCs enclosing the region and whether it was found to be a predictor of prognosis (one color for good prognosis and another for poor prognosis). For example, FANCD2, is mutated in all of the SCC types except lung SCC, as can be seen at the end of this video: https://www.youtube.com/embed/p8C3DEN2Ah4\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "source": [
    "### Progress\n",
    "\n",
    "Thus far, I used R to carry out the analysis. The wrangling and analysis was also performed solely on genomic data (mutated genes and specific mutations), while patient data will need to be added for the supervised machine learning studies. The R code has been deposited in the parent Github repository for this project (ejdna/capstone2/tcga.R). This R script also contains some basic unsupervised machine learning analysis (heirarchical and k-means clustering). This analysis was carried out at the level of the individual studies not at the level of individual patients, the latter being the more appropriate level and the focus of subsequent analysis. The dendrogram from the heirarchical clustering can be found here: ejdna/capstone2/Hclust_scc.jpeg, and pie charts showing the proportion of studies in different clusters from the k-means clustering for different values of k are generated as plots when the script is run (presuming all of the packages are properly updated, which may be a significant problem in itself). For subsequent analysis involving supervised machine learning, I will need to take advantage of the sklearn and TensorFlow packages in Python, which means 1) transferring the cleaned mutation data from R to Pandas and 2) transferring the patient data to Python and preparing it for the machine learning pipeline. \n",
    "\n",
    "#### Genomic mutation data\n",
    "\n",
    "The following R script was used to write the cleaned dataframes from R to csv files: ejdna/capstone2/writeRdataframes2file.R (The <study>_Routput.csf files in this directory are the csv files generated by this script. These dataframes contain only columns that would be of potential interest for the proposed analysis and only rows containing mutations that are predicted to have moderate to high impact on the function of the mutated genes. The notations I used for the studies follows:\n",
    "     \n",
    "    Cervical SCC:      cervical  (unpublished TCGA data)\n",
    "    Esophageal SCC:    esoph   (UCLA study, 2014)\n",
    "                       esoph2  (ICGC study, 2014)\n",
    "    Head and Neck SCC: hn   (Broad study, 2011)\n",
    "                       hn2  (Hopkins study, 2011)\n",
    "                       hn3  (MDA study, 2013)\n",
    "                       hn4  (TCGA published data, 2015)\n",
    "                       hn5  (unpublished TCGA data)\n",
    "    Lung SCC:          lung  (TCGA published data, 2012)\n",
    "                       lung2 (unpublished TCGA data)\n",
    "    Skin SCC:          skin  (Dana Farber study, 2015)\n",
    "\n",
    "Of these, only cervical, esoph2, hn3, hn4, hn5, lung2 and skin also have patient data files that contained the overall survival information needed for training and testing the supervised machine learning models. Thus, these are the datasets I will focus on going forward.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Gene</th>\n",
       "      <th>VariantClass</th>\n",
       "      <th>VariantType</th>\n",
       "      <th>SampleBC</th>\n",
       "      <th>MutStat</th>\n",
       "      <th>SeqSource</th>\n",
       "      <th>SampleID</th>\n",
       "      <th>DNAchange</th>\n",
       "      <th>ProteinChange</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>Impact</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>ABCA10</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>TCGA-BI-A0VR-01</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>WXS</td>\n",
       "      <td>dc62866c-0dc0-4ef5-bc63-fee1e943a99f</td>\n",
       "      <td>c.2823G&gt;C</td>\n",
       "      <td>p.L941F</td>\n",
       "      <td>tolerated(0.47)</td>\n",
       "      <td>benign(0.029)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>AIM1</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>TCGA-BI-A0VR-01</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>WXS</td>\n",
       "      <td>dc62866c-0dc0-4ef5-bc63-fee1e943a99f</td>\n",
       "      <td>c.1216G&gt;C</td>\n",
       "      <td>p.E406Q</td>\n",
       "      <td>tolerated_low_confidence(0.1)</td>\n",
       "      <td>benign(0.048)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>AK7</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>TCGA-BI-A0VR-01</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>WXS</td>\n",
       "      <td>dc62866c-0dc0-4ef5-bc63-fee1e943a99f</td>\n",
       "      <td>c.76G&gt;T</td>\n",
       "      <td>p.D26Y</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably_damaging(1)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>ANO9</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>TCGA-BI-A0VR-01</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>WXS</td>\n",
       "      <td>dc62866c-0dc0-4ef5-bc63-fee1e943a99f</td>\n",
       "      <td>c.292G&gt;A</td>\n",
       "      <td>p.E98K</td>\n",
       "      <td>deleterious(0.02)</td>\n",
       "      <td>benign(0.02)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>ARID4A</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>TCGA-BI-A0VR-01</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>WXS</td>\n",
       "      <td>dc62866c-0dc0-4ef5-bc63-fee1e943a99f</td>\n",
       "      <td>c.1525A&gt;G</td>\n",
       "      <td>p.T509A</td>\n",
       "      <td>tolerated(0.26)</td>\n",
       "      <td>benign(0.001)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0    Gene       VariantClass VariantType         SampleBC  \\\n",
       "0           1  ABCA10  Missense_Mutation         SNP  TCGA-BI-A0VR-01   \n",
       "1           2    AIM1  Missense_Mutation         SNP  TCGA-BI-A0VR-01   \n",
       "2           3     AK7  Missense_Mutation         SNP  TCGA-BI-A0VR-01   \n",
       "3           4    ANO9  Missense_Mutation         SNP  TCGA-BI-A0VR-01   \n",
       "4           5  ARID4A  Missense_Mutation         SNP  TCGA-BI-A0VR-01   \n",
       "\n",
       "   MutStat SeqSource                              SampleID  DNAchange  \\\n",
       "0  Somatic       WXS  dc62866c-0dc0-4ef5-bc63-fee1e943a99f  c.2823G>C   \n",
       "1  Somatic       WXS  dc62866c-0dc0-4ef5-bc63-fee1e943a99f  c.1216G>C   \n",
       "2  Somatic       WXS  dc62866c-0dc0-4ef5-bc63-fee1e943a99f    c.76G>T   \n",
       "3  Somatic       WXS  dc62866c-0dc0-4ef5-bc63-fee1e943a99f   c.292G>A   \n",
       "4  Somatic       WXS  dc62866c-0dc0-4ef5-bc63-fee1e943a99f  c.1525A>G   \n",
       "\n",
       "  ProteinChange                           SIFT              PolyPhen    Impact  \n",
       "0       p.L941F                tolerated(0.47)         benign(0.029)  MODERATE  \n",
       "1       p.E406Q  tolerated_low_confidence(0.1)         benign(0.048)  MODERATE  \n",
       "2        p.D26Y                 deleterious(0)  probably_damaging(1)  MODERATE  \n",
       "3        p.E98K              deleterious(0.02)          benign(0.02)  MODERATE  \n",
       "4       p.T509A                tolerated(0.26)         benign(0.001)  MODERATE  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# load data from csv files into pandas\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re\n",
    "\n",
    "cervical_genomic_df = pd.read_csv(\"../cervical_Routput.csv\")\n",
    "cervical_genomic_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Gene</th>\n",
       "      <th>VariantClass</th>\n",
       "      <th>VariantType</th>\n",
       "      <th>SampleBC</th>\n",
       "      <th>MutStat</th>\n",
       "      <th>SeqSource</th>\n",
       "      <th>SampleID</th>\n",
       "      <th>DNAchange</th>\n",
       "      <th>ProteinChange</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>Impact</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>CACNA1E</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>ESCC-001T</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.327G&gt;T</td>\n",
       "      <td>p.E109D</td>\n",
       "      <td>deleterious(0.02)</td>\n",
       "      <td>benign(0.217)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>SP140</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>ESCC-001T</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.934G&gt;C</td>\n",
       "      <td>p.E312Q</td>\n",
       "      <td>deleterious(0.04)</td>\n",
       "      <td>probably_damaging(0.974)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>PEX5L</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>ESCC-001T</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.64G&gt;A</td>\n",
       "      <td>p.D22N</td>\n",
       "      <td>deleterious_low_confidence(0)</td>\n",
       "      <td>benign(0.062)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>MATR3</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>ESCC-001T</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.1306G&gt;A</td>\n",
       "      <td>p.E436K</td>\n",
       "      <td>tolerated(0.59)</td>\n",
       "      <td>benign(0.013)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>ABCA13</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>ESCC-001T</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.847G&gt;C</td>\n",
       "      <td>p.D283H</td>\n",
       "      <td>NaN</td>\n",
       "      <td>probably_damaging(0.917)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0     Gene       VariantClass VariantType   SampleBC  MutStat  \\\n",
       "0           1  CACNA1E  Missense_Mutation         SNP  ESCC-001T  Somatic   \n",
       "1           2    SP140  Missense_Mutation         SNP  ESCC-001T  Somatic   \n",
       "2           3    PEX5L  Missense_Mutation         SNP  ESCC-001T  Somatic   \n",
       "3           4    MATR3  Missense_Mutation         SNP  ESCC-001T  Somatic   \n",
       "4           5   ABCA13  Missense_Mutation         SNP  ESCC-001T  Somatic   \n",
       "\n",
       "   SeqSource  SampleID  DNAchange ProteinChange  \\\n",
       "0        NaN       NaN   c.327G>T       p.E109D   \n",
       "1        NaN       NaN   c.934G>C       p.E312Q   \n",
       "2        NaN       NaN    c.64G>A        p.D22N   \n",
       "3        NaN       NaN  c.1306G>A       p.E436K   \n",
       "4        NaN       NaN   c.847G>C       p.D283H   \n",
       "\n",
       "                            SIFT                  PolyPhen    Impact  \n",
       "0              deleterious(0.02)             benign(0.217)  MODERATE  \n",
       "1              deleterious(0.04)  probably_damaging(0.974)  MODERATE  \n",
       "2  deleterious_low_confidence(0)             benign(0.062)  MODERATE  \n",
       "3                tolerated(0.59)             benign(0.013)  MODERATE  \n",
       "4                            NaN  probably_damaging(0.917)  MODERATE  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "esoph2_genomic_df = pd.read_csv(\"../esoph2_Routput.csv\")\n",
    "esoph2_genomic_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Gene</th>\n",
       "      <th>VariantClass</th>\n",
       "      <th>VariantType</th>\n",
       "      <th>SampleBC</th>\n",
       "      <th>MutStat</th>\n",
       "      <th>SeqSource</th>\n",
       "      <th>SampleID</th>\n",
       "      <th>DNAchange</th>\n",
       "      <th>ProteinChange</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>Impact</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>OSCJM-PT01-166-T</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>Capture</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.155A&gt;G</td>\n",
       "      <td>p.H52R</td>\n",
       "      <td>tolerated(0.91)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>ABCF3</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>OSCJM-PT01-166-T</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>Capture</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.1123T&gt;C</td>\n",
       "      <td>p.S375P</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>possibly_damaging(0.866)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>ACAD10</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>OSCJM-PT01-166-T</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>Capture</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.1259G&gt;T</td>\n",
       "      <td>p.R420L</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>benign(0.278)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>ACSF3</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>OSCJM-PT01-166-T</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>Capture</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.1114G&gt;A</td>\n",
       "      <td>p.V372M</td>\n",
       "      <td>tolerated(0.1)</td>\n",
       "      <td>possibly_damaging(0.631)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>ADAM21</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>OSCJM-PT01-166-T</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>Capture</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.214C&gt;T</td>\n",
       "      <td>p.H72Y</td>\n",
       "      <td>tolerated(0.06)</td>\n",
       "      <td>benign(0.107)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0    Gene       VariantClass VariantType          SampleBC  \\\n",
       "0           1    A1BG  Missense_Mutation         SNP  OSCJM-PT01-166-T   \n",
       "1           2   ABCF3  Missense_Mutation         SNP  OSCJM-PT01-166-T   \n",
       "2           3  ACAD10  Missense_Mutation         SNP  OSCJM-PT01-166-T   \n",
       "3           4   ACSF3  Missense_Mutation         SNP  OSCJM-PT01-166-T   \n",
       "4           5  ADAM21  Missense_Mutation         SNP  OSCJM-PT01-166-T   \n",
       "\n",
       "   MutStat SeqSource  SampleID  DNAchange ProteinChange             SIFT  \\\n",
       "0  Somatic   Capture       NaN   c.155A>G        p.H52R  tolerated(0.91)   \n",
       "1  Somatic   Capture       NaN  c.1123T>C       p.S375P   deleterious(0)   \n",
       "2  Somatic   Capture       NaN  c.1259G>T       p.R420L   deleterious(0)   \n",
       "3  Somatic   Capture       NaN  c.1114G>A       p.V372M   tolerated(0.1)   \n",
       "4  Somatic   Capture       NaN   c.214C>T        p.H72Y  tolerated(0.06)   \n",
       "\n",
       "                   PolyPhen    Impact  \n",
       "0                 benign(0)  MODERATE  \n",
       "1  possibly_damaging(0.866)  MODERATE  \n",
       "2             benign(0.278)  MODERATE  \n",
       "3  possibly_damaging(0.631)  MODERATE  \n",
       "4             benign(0.107)  MODERATE  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hn3_genomic_df = pd.read_csv(\"../hn3_Routput.csv\")\n",
    "hn3_genomic_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Gene</th>\n",
       "      <th>VariantClass</th>\n",
       "      <th>VariantType</th>\n",
       "      <th>SampleBC</th>\n",
       "      <th>MutStat</th>\n",
       "      <th>SeqSource</th>\n",
       "      <th>SampleID</th>\n",
       "      <th>DNAchange</th>\n",
       "      <th>ProteinChange</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>Impact</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>SLC2A7</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>TCGA-BA-4074-01-01-SM-1PNIE</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unspecified</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.17C&gt;A</td>\n",
       "      <td>p.A6E</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>VPS13D</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>TCGA-BA-4074-01-01-SM-1PNIE</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unspecified</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.10297A&gt;T</td>\n",
       "      <td>p.I3433F</td>\n",
       "      <td>NaN</td>\n",
       "      <td>benign(0.076)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>SZT2</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>TCGA-BA-4074-01-01-SM-1PNIE</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unspecified</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.7898A&gt;T</td>\n",
       "      <td>p.H2633L</td>\n",
       "      <td>deleterious(0.01)</td>\n",
       "      <td>probably_damaging(0.997)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>LPHN2</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>TCGA-BA-4074-01-01-SM-1PNIE</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unspecified</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.2777C&gt;A</td>\n",
       "      <td>p.A926D</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>benign(0.169)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>ARHGAP29</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>TCGA-BA-4074-01-01-SM-1PNIE</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unspecified</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.2367G&gt;C</td>\n",
       "      <td>p.W789C</td>\n",
       "      <td>tolerated(0.18)</td>\n",
       "      <td>benign(0.006)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0      Gene       VariantClass VariantType  \\\n",
       "0           1    SLC2A7  Missense_Mutation         SNP   \n",
       "1           2    VPS13D  Missense_Mutation         SNP   \n",
       "2           3      SZT2  Missense_Mutation         SNP   \n",
       "3           4     LPHN2  Missense_Mutation         SNP   \n",
       "4           5  ARHGAP29  Missense_Mutation         SNP   \n",
       "\n",
       "                      SampleBC  MutStat    SeqSource  SampleID   DNAchange  \\\n",
       "0  TCGA-BA-4074-01-01-SM-1PNIE  Unknown  Unspecified       NaN     c.17C>A   \n",
       "1  TCGA-BA-4074-01-01-SM-1PNIE  Unknown  Unspecified       NaN  c.10297A>T   \n",
       "2  TCGA-BA-4074-01-01-SM-1PNIE  Unknown  Unspecified       NaN   c.7898A>T   \n",
       "3  TCGA-BA-4074-01-01-SM-1PNIE  Unknown  Unspecified       NaN   c.2777C>A   \n",
       "4  TCGA-BA-4074-01-01-SM-1PNIE  Unknown  Unspecified       NaN   c.2367G>C   \n",
       "\n",
       "  ProteinChange               SIFT                  PolyPhen    Impact  \n",
       "0         p.A6E       tolerated(1)                 benign(0)  MODERATE  \n",
       "1      p.I3433F                NaN             benign(0.076)  MODERATE  \n",
       "2      p.H2633L  deleterious(0.01)  probably_damaging(0.997)  MODERATE  \n",
       "3       p.A926D     deleterious(0)             benign(0.169)  MODERATE  \n",
       "4       p.W789C    tolerated(0.18)             benign(0.006)  MODERATE  "
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hn4_genomic_df = pd.read_csv(\"../hn4_Routput.csv\")\n",
    "hn4_genomic_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Gene</th>\n",
       "      <th>VariantClass</th>\n",
       "      <th>VariantType</th>\n",
       "      <th>SampleBC</th>\n",
       "      <th>MutStat</th>\n",
       "      <th>SeqSource</th>\n",
       "      <th>SampleID</th>\n",
       "      <th>DNAchange</th>\n",
       "      <th>ProteinChange</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>Impact</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>ABCA13</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>TCGA-4P-AA8J-01</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>WXS</td>\n",
       "      <td>0fe5f8b2-c794-4c93-b2ab-0b544e366f5c</td>\n",
       "      <td>c.6639C&gt;G</td>\n",
       "      <td>p.I2213M</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>possibly_damaging(0.453)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>ZMAT2</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>TCGA-4P-AA8J-01</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>WXS</td>\n",
       "      <td>0fe5f8b2-c794-4c93-b2ab-0b544e366f5c</td>\n",
       "      <td>c.363G&gt;C</td>\n",
       "      <td>p.Q121H</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>possibly_damaging(0.764)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>APPL2</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>TCGA-4P-AA8J-01</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>WXS</td>\n",
       "      <td>0fe5f8b2-c794-4c93-b2ab-0b544e366f5c</td>\n",
       "      <td>c.1261G&gt;C</td>\n",
       "      <td>p.E421Q</td>\n",
       "      <td>tolerated(0.42)</td>\n",
       "      <td>benign(0.011)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>LRSAM1</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>TCGA-4P-AA8J-01</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>WXS</td>\n",
       "      <td>0fe5f8b2-c794-4c93-b2ab-0b544e366f5c</td>\n",
       "      <td>c.825A&gt;C</td>\n",
       "      <td>p.E275D</td>\n",
       "      <td>tolerated(0.51)</td>\n",
       "      <td>benign(0.002)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>ZNF425</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>TCGA-4P-AA8J-01</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>WXS</td>\n",
       "      <td>0fe5f8b2-c794-4c93-b2ab-0b544e366f5c</td>\n",
       "      <td>c.276G&gt;A</td>\n",
       "      <td>p.M92I</td>\n",
       "      <td>tolerated(0.07)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0    Gene       VariantClass VariantType         SampleBC  \\\n",
       "0           1  ABCA13  Missense_Mutation         SNP  TCGA-4P-AA8J-01   \n",
       "1           2   ZMAT2  Missense_Mutation         SNP  TCGA-4P-AA8J-01   \n",
       "2           3   APPL2  Missense_Mutation         SNP  TCGA-4P-AA8J-01   \n",
       "3           4  LRSAM1  Missense_Mutation         SNP  TCGA-4P-AA8J-01   \n",
       "4           5  ZNF425  Missense_Mutation         SNP  TCGA-4P-AA8J-01   \n",
       "\n",
       "   MutStat SeqSource                              SampleID  DNAchange  \\\n",
       "0  Somatic       WXS  0fe5f8b2-c794-4c93-b2ab-0b544e366f5c  c.6639C>G   \n",
       "1  Somatic       WXS  0fe5f8b2-c794-4c93-b2ab-0b544e366f5c   c.363G>C   \n",
       "2  Somatic       WXS  0fe5f8b2-c794-4c93-b2ab-0b544e366f5c  c.1261G>C   \n",
       "3  Somatic       WXS  0fe5f8b2-c794-4c93-b2ab-0b544e366f5c   c.825A>C   \n",
       "4  Somatic       WXS  0fe5f8b2-c794-4c93-b2ab-0b544e366f5c   c.276G>A   \n",
       "\n",
       "  ProteinChange             SIFT                  PolyPhen    Impact  \n",
       "0      p.I2213M  [Not Available]  possibly_damaging(0.453)  MODERATE  \n",
       "1       p.Q121H   deleterious(0)  possibly_damaging(0.764)  MODERATE  \n",
       "2       p.E421Q  tolerated(0.42)             benign(0.011)  MODERATE  \n",
       "3       p.E275D  tolerated(0.51)             benign(0.002)  MODERATE  \n",
       "4        p.M92I  tolerated(0.07)                 benign(0)  MODERATE  "
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hn5_genomic_df = pd.read_csv(\"../hn5_Routput.csv\")\n",
    "hn5_genomic_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Gene</th>\n",
       "      <th>VariantClass</th>\n",
       "      <th>VariantType</th>\n",
       "      <th>SampleBC</th>\n",
       "      <th>MutStat</th>\n",
       "      <th>SeqSource</th>\n",
       "      <th>SampleID</th>\n",
       "      <th>DNAchange</th>\n",
       "      <th>ProteinChange</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>Impact</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>ANKFN1</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>TCGA-18-3406-01</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>Capture</td>\n",
       "      <td>d3320989-71fd-425b-933e-6e8528a016ed</td>\n",
       "      <td>c.840G&gt;T</td>\n",
       "      <td>p.M280I</td>\n",
       "      <td>tolerated(0.4)</td>\n",
       "      <td>benign(0.005)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>TCGA-18-3406-01</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>Capture</td>\n",
       "      <td>d3320989-71fd-425b-933e-6e8528a016ed</td>\n",
       "      <td>c.2705C&gt;T</td>\n",
       "      <td>p.A902V</td>\n",
       "      <td>tolerated(0.25)</td>\n",
       "      <td>possibly_damaging(0.499)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>GLTPD1</td>\n",
       "      <td>Nonstop_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>TCGA-18-3406-01</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>Capture</td>\n",
       "      <td>d3320989-71fd-425b-933e-6e8528a016ed</td>\n",
       "      <td>c.645G&gt;C</td>\n",
       "      <td>p.*215Yext*44</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>HIGH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>ACTRT2</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>TCGA-18-3406-01</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>Capture</td>\n",
       "      <td>d3320989-71fd-425b-933e-6e8528a016ed</td>\n",
       "      <td>c.1095G&gt;T</td>\n",
       "      <td>p.K365N</td>\n",
       "      <td>tolerated(0.16)</td>\n",
       "      <td>benign(0.072)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>DCDC2B</td>\n",
       "      <td>Nonsense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>TCGA-18-3406-01</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>Capture</td>\n",
       "      <td>d3320989-71fd-425b-933e-6e8528a016ed</td>\n",
       "      <td>c.689C&gt;A</td>\n",
       "      <td>p.S230*</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>HIGH</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0    Gene       VariantClass VariantType         SampleBC  \\\n",
       "0           1  ANKFN1  Missense_Mutation         SNP  TCGA-18-3406-01   \n",
       "1           2    AGRN  Missense_Mutation         SNP  TCGA-18-3406-01   \n",
       "2           3  GLTPD1   Nonstop_Mutation         SNP  TCGA-18-3406-01   \n",
       "3           4  ACTRT2  Missense_Mutation         SNP  TCGA-18-3406-01   \n",
       "4           5  DCDC2B  Nonsense_Mutation         SNP  TCGA-18-3406-01   \n",
       "\n",
       "   MutStat SeqSource                              SampleID  DNAchange  \\\n",
       "0  Somatic   Capture  d3320989-71fd-425b-933e-6e8528a016ed   c.840G>T   \n",
       "1  Somatic   Capture  d3320989-71fd-425b-933e-6e8528a016ed  c.2705C>T   \n",
       "2  Somatic   Capture  d3320989-71fd-425b-933e-6e8528a016ed   c.645G>C   \n",
       "3  Somatic   Capture  d3320989-71fd-425b-933e-6e8528a016ed  c.1095G>T   \n",
       "4  Somatic   Capture  d3320989-71fd-425b-933e-6e8528a016ed   c.689C>A   \n",
       "\n",
       "   ProteinChange             SIFT                  PolyPhen    Impact  \n",
       "0        p.M280I   tolerated(0.4)             benign(0.005)  MODERATE  \n",
       "1        p.A902V  tolerated(0.25)  possibly_damaging(0.499)  MODERATE  \n",
       "2  p.*215Yext*44  [Not Available]           [Not Available]      HIGH  \n",
       "3        p.K365N  tolerated(0.16)             benign(0.072)  MODERATE  \n",
       "4        p.S230*  [Not Available]           [Not Available]      HIGH  "
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lung2_genomic_df = pd.read_csv(\"../lung2_Routput.csv\")\n",
    "lung2_genomic_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Gene</th>\n",
       "      <th>VariantClass</th>\n",
       "      <th>VariantType</th>\n",
       "      <th>SampleBC</th>\n",
       "      <th>MutStat</th>\n",
       "      <th>SeqSource</th>\n",
       "      <th>SampleID</th>\n",
       "      <th>DNAchange</th>\n",
       "      <th>ProteinChange</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>Impact</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>AFF3</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>S00-28455-TP-NT</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>WXS</td>\n",
       "      <td>S00-28455-TP</td>\n",
       "      <td>c.1082C&gt;T</td>\n",
       "      <td>p.S361L</td>\n",
       "      <td>tolerated(0.12)</td>\n",
       "      <td>benign(0.251)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>AGTR1</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>S00-28455-TP-NT</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>WXS</td>\n",
       "      <td>S00-28455-TP</td>\n",
       "      <td>c.377G&gt;T</td>\n",
       "      <td>p.R126L</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably_damaging(1)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>ARID2</td>\n",
       "      <td>Nonsense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>S00-28455-TP-NT</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>WXS</td>\n",
       "      <td>S00-28455-TP</td>\n",
       "      <td>c.797G&gt;A</td>\n",
       "      <td>p.W266*</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HIGH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>AXIN2</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>S00-28455-TP-NT</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>WXS</td>\n",
       "      <td>S00-28455-TP</td>\n",
       "      <td>c.290C&gt;T</td>\n",
       "      <td>p.T97I</td>\n",
       "      <td>deleterious(0.01)</td>\n",
       "      <td>benign(0.077)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>S00-28455-TP-NT</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>WXS</td>\n",
       "      <td>S00-28455-TP</td>\n",
       "      <td>c.3575C&gt;T</td>\n",
       "      <td>p.P1192L</td>\n",
       "      <td>deleterious(0.02)</td>\n",
       "      <td>benign(0.391)</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0   Gene       VariantClass VariantType         SampleBC  MutStat  \\\n",
       "0           1   AFF3  Missense_Mutation         SNP  S00-28455-TP-NT  Somatic   \n",
       "1           2  AGTR1  Missense_Mutation         SNP  S00-28455-TP-NT  Somatic   \n",
       "2           3  ARID2  Nonsense_Mutation         SNP  S00-28455-TP-NT  Somatic   \n",
       "3           4  AXIN2  Missense_Mutation         SNP  S00-28455-TP-NT  Somatic   \n",
       "4           5  BRCA1  Missense_Mutation         SNP  S00-28455-TP-NT  Somatic   \n",
       "\n",
       "  SeqSource      SampleID  DNAchange ProteinChange               SIFT  \\\n",
       "0       WXS  S00-28455-TP  c.1082C>T       p.S361L    tolerated(0.12)   \n",
       "1       WXS  S00-28455-TP   c.377G>T       p.R126L     deleterious(0)   \n",
       "2       WXS  S00-28455-TP   c.797G>A       p.W266*                NaN   \n",
       "3       WXS  S00-28455-TP   c.290C>T        p.T97I  deleterious(0.01)   \n",
       "4       WXS  S00-28455-TP  c.3575C>T      p.P1192L  deleterious(0.02)   \n",
       "\n",
       "               PolyPhen    Impact  \n",
       "0         benign(0.251)  MODERATE  \n",
       "1  probably_damaging(1)  MODERATE  \n",
       "2                   NaN      HIGH  \n",
       "3         benign(0.077)  MODERATE  \n",
       "4         benign(0.391)  MODERATE  "
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "skin_genomic_df = pd.read_csv(\"../skin_Routput.csv\")\n",
    "skin_genomic_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Patient Data\n",
    "\n",
    "The patient data was also downloaded from TCGA and added to the notebook. The following code populates Pandas dataframes with the relevant data (chosen columns provide data that is in common to most or all of the datasets)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PATIENT_ID</th>\n",
       "      <th>GENDER</th>\n",
       "      <th>TOBACCO_SMOKING_HISTORY_INDICATOR</th>\n",
       "      <th>SMOKING_YEAR_STARTED</th>\n",
       "      <th>SMOKING_YEAR_STOPPED</th>\n",
       "      <th>SMOKING_PACK_YEARS</th>\n",
       "      <th>AGE</th>\n",
       "      <th>HISTOLOGICAL_DIAGNOSIS</th>\n",
       "      <th>GRADE</th>\n",
       "      <th>INITIAL_PATHOLOGIC_DX_YEAR</th>\n",
       "      <th>AJCC_NODES_PATHOLOGIC_PN</th>\n",
       "      <th>AJCC_TUMOR_PATHOLOGIC_PT</th>\n",
       "      <th>AJCC_METASTASIS_PATHOLOGIC_PM</th>\n",
       "      <th>CLINICAL_STAGE</th>\n",
       "      <th>TUMOR_TISSUE_SITE</th>\n",
       "      <th>OS_STATUS</th>\n",
       "      <th>OS_MONTHS</th>\n",
       "      <th>DFS_STATUS</th>\n",
       "      <th>DFS_MONTHS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-4J-AA1J</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>1</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>31</td>\n",
       "      <td>Cervical Squamous Cell Carcinoma</td>\n",
       "      <td>G3</td>\n",
       "      <td>2013</td>\n",
       "      <td>N0</td>\n",
       "      <td>T1b2</td>\n",
       "      <td>M0</td>\n",
       "      <td>Stage IB2</td>\n",
       "      <td>Cervical</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>17.81</td>\n",
       "      <td>DiseaseFree</td>\n",
       "      <td>17.81</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-BI-A0VR</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>4</td>\n",
       "      <td>22</td>\n",
       "      <td>1995</td>\n",
       "      <td>20</td>\n",
       "      <td>53</td>\n",
       "      <td>Cervical Squamous Cell Carcinoma</td>\n",
       "      <td>G3</td>\n",
       "      <td>2006</td>\n",
       "      <td>N1</td>\n",
       "      <td>T2b</td>\n",
       "      <td>M0</td>\n",
       "      <td>Stage IIIB</td>\n",
       "      <td>Cervical</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>49.44</td>\n",
       "      <td>DiseaseFree</td>\n",
       "      <td>49.44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-BI-A0VS</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>2</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>48</td>\n",
       "      <td>Cervical Squamous Cell Carcinoma</td>\n",
       "      <td>G3</td>\n",
       "      <td>2007</td>\n",
       "      <td>N0</td>\n",
       "      <td>T1b1</td>\n",
       "      <td>M0</td>\n",
       "      <td>Stage IB</td>\n",
       "      <td>Cervical</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>57.00</td>\n",
       "      <td>DiseaseFree</td>\n",
       "      <td>57.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-BI-A20A</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>49</td>\n",
       "      <td>Cervical Squamous Cell Carcinoma</td>\n",
       "      <td>G3</td>\n",
       "      <td>2010</td>\n",
       "      <td>N0</td>\n",
       "      <td>T1b1</td>\n",
       "      <td>M0</td>\n",
       "      <td>Stage IB1</td>\n",
       "      <td>Cervical</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>23.65</td>\n",
       "      <td>DiseaseFree</td>\n",
       "      <td>23.65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-C5-A0TN</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>1</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>21</td>\n",
       "      <td>Cervical Squamous Cell Carcinoma</td>\n",
       "      <td>G3</td>\n",
       "      <td>1997</td>\n",
       "      <td>N1</td>\n",
       "      <td>T1b</td>\n",
       "      <td>MX</td>\n",
       "      <td>Stage IB2</td>\n",
       "      <td>Cervical</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>11.43</td>\n",
       "      <td>Recurred/Progressed</td>\n",
       "      <td>2.04</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     PATIENT_ID  GENDER TOBACCO_SMOKING_HISTORY_INDICATOR  \\\n",
       "0  TCGA-4J-AA1J  FEMALE                                 1   \n",
       "1  TCGA-BI-A0VR  FEMALE                                 4   \n",
       "2  TCGA-BI-A0VS  FEMALE                                 2   \n",
       "3  TCGA-BI-A20A  FEMALE                   [Not Available]   \n",
       "4  TCGA-C5-A0TN  FEMALE                                 1   \n",
       "\n",
       "  SMOKING_YEAR_STARTED SMOKING_YEAR_STOPPED SMOKING_PACK_YEARS  AGE  \\\n",
       "0      [Not Available]      [Not Available]    [Not Available]   31   \n",
       "1                   22                 1995                 20   53   \n",
       "2      [Not Available]      [Not Available]    [Not Available]   48   \n",
       "3      [Not Available]      [Not Available]    [Not Available]   49   \n",
       "4      [Not Available]      [Not Available]    [Not Available]   21   \n",
       "\n",
       "             HISTOLOGICAL_DIAGNOSIS GRADE INITIAL_PATHOLOGIC_DX_YEAR  \\\n",
       "0  Cervical Squamous Cell Carcinoma    G3                       2013   \n",
       "1  Cervical Squamous Cell Carcinoma    G3                       2006   \n",
       "2  Cervical Squamous Cell Carcinoma    G3                       2007   \n",
       "3  Cervical Squamous Cell Carcinoma    G3                       2010   \n",
       "4  Cervical Squamous Cell Carcinoma    G3                       1997   \n",
       "\n",
       "  AJCC_NODES_PATHOLOGIC_PN AJCC_TUMOR_PATHOLOGIC_PT  \\\n",
       "0                       N0                     T1b2   \n",
       "1                       N1                      T2b   \n",
       "2                       N0                     T1b1   \n",
       "3                       N0                     T1b1   \n",
       "4                       N1                      T1b   \n",
       "\n",
       "  AJCC_METASTASIS_PATHOLOGIC_PM CLINICAL_STAGE TUMOR_TISSUE_SITE OS_STATUS  \\\n",
       "0                            M0      Stage IB2          Cervical    LIVING   \n",
       "1                            M0     Stage IIIB          Cervical    LIVING   \n",
       "2                            M0       Stage IB          Cervical    LIVING   \n",
       "3                            M0      Stage IB1          Cervical    LIVING   \n",
       "4                            MX      Stage IB2          Cervical  DECEASED   \n",
       "\n",
       "   OS_MONTHS           DFS_STATUS DFS_MONTHS  \n",
       "0      17.81          DiseaseFree      17.81  \n",
       "1      49.44          DiseaseFree      49.44  \n",
       "2      57.00          DiseaseFree      57.00  \n",
       "3      23.65          DiseaseFree      23.65  \n",
       "4      11.43  Recurred/Progressed       2.04  "
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Now get the correspending patient data for each dataset\n",
    "#The patient data files are in the same directory as the R output files\n",
    "cervical_patient_df = pd.read_table(\"../cervical_data_bcr_clinical_data_patient.txt\", usecols = lambda x: x in [\n",
    "                                                                                               \"PATIENT_ID\", \n",
    "                                                                                               \"GENDER\", \n",
    "                                                                                               \"TOBACCO_SMOKING_HISTORY_INDICATOR\", \n",
    "                                                                                               \"SMOKING_YEAR_STARTED\", \n",
    "                                                                                               \"SMOKING_YEAR_STOPPED\",\n",
    "                                                                                               \"SMOKING_PACK_YEARS\",\n",
    "                                                                                               \"AGE\",\n",
    "                                                                                               \"HISTOLOGICAL_DIAGNOSIS\",\n",
    "                                                                                               \"GRADE\",\n",
    "                                                                                               \"INITIAL_PATHOLOGIC_DX_YEAR\", \n",
    "                                                                                               \"AJCC_NODES_PATHOLOGIC_PN\",\n",
    "                                                                                               \"AJCC_TUMOR_PATHOLOGIC_PT\",\n",
    "                                                                                               \"AJCC_METASTASIS_PATHOLOGIC_PM\",\n",
    "                                                                                               \"CLINICAL_STAGE\", \n",
    "                                                                                               \"TUMOR_TISSUE_SITE\",\n",
    "                                                                                               \"OS_STATUS\",\n",
    "                                                                                               \"OS_MONTHS\",\n",
    "                                                                                               \"DFS_STATUS\",\n",
    "                                                                                               \"DFS_MONTHS\"],\n",
    "                                   skiprows = 4)\n",
    "cervical_patient_df.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SAMPLE_ID</th>\n",
       "      <th>PATIENT_ID</th>\n",
       "      <th>GENDER</th>\n",
       "      <th>AGE</th>\n",
       "      <th>OS_STATUS</th>\n",
       "      <th>OS_MONTHS</th>\n",
       "      <th>TUMOR_STAGE</th>\n",
       "      <th>TNM</th>\n",
       "      <th>SMOKER</th>\n",
       "      <th>CANCER_TYPE_DETAILED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ESCC-001T</td>\n",
       "      <td>ESCC-001T</td>\n",
       "      <td>Male</td>\n",
       "      <td>74</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>24.61</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>T2N2M0</td>\n",
       "      <td>No</td>\n",
       "      <td>Esophageal Squamous Cell Carcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ESCC-002T</td>\n",
       "      <td>ESCC-002T</td>\n",
       "      <td>Male</td>\n",
       "      <td>47</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>7.79</td>\n",
       "      <td>IIB</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Esophageal Squamous Cell Carcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ESCC-003T</td>\n",
       "      <td>ESCC-003T</td>\n",
       "      <td>Female</td>\n",
       "      <td>54</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>45.37</td>\n",
       "      <td>IIB</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>No</td>\n",
       "      <td>Esophageal Squamous Cell Carcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ESCC-004T</td>\n",
       "      <td>ESCC-004T</td>\n",
       "      <td>Male</td>\n",
       "      <td>60</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>15.41</td>\n",
       "      <td>IIB</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>No</td>\n",
       "      <td>Esophageal Squamous Cell Carcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ESCC-005T</td>\n",
       "      <td>ESCC-005T</td>\n",
       "      <td>Male</td>\n",
       "      <td>65</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>29.37</td>\n",
       "      <td>IIB</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Esophageal Squamous Cell Carcinoma</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   SAMPLE_ID PATIENT_ID  GENDER  AGE OS_STATUS  OS_MONTHS TUMOR_STAGE  \\\n",
       "0  ESCC-001T  ESCC-001T    Male   74    LIVING      24.61        IIIA   \n",
       "1  ESCC-002T  ESCC-002T    Male   47  DECEASED       7.79         IIB   \n",
       "2  ESCC-003T  ESCC-003T  Female   54    LIVING      45.37         IIB   \n",
       "3  ESCC-004T  ESCC-004T    Male   60  DECEASED      15.41         IIB   \n",
       "4  ESCC-005T  ESCC-005T    Male   65  DECEASED      29.37         IIB   \n",
       "\n",
       "       TNM SMOKER                CANCER_TYPE_DETAILED  \n",
       "0   T2N2M0     No  Esophageal Squamous Cell Carcinoma  \n",
       "1   T2N0M0    Yes  Esophageal Squamous Cell Carcinoma  \n",
       "2   T2N1M0     No  Esophageal Squamous Cell Carcinoma  \n",
       "3   T2N0M0     No  Esophageal Squamous Cell Carcinoma  \n",
       "4   T2N0M0    Yes  Esophageal Squamous Cell Carcinoma  "
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "esoph2_patient_df = pd.read_table(\"../esoph2_data_clinical.txt\", usecols = lambda x: x in  [\"SAMPLE_ID\",\n",
    "                                                                                            \"PATIENT_ID\", \n",
    "                                                                                            \"GENDER\", \n",
    "                                                                                            \"AGE\",\n",
    "                                                                                            \"OS_STATUS\",\n",
    "                                                                                            \"OS_MONTHS\",\n",
    "                                                                                            \"TUMOR_STAGE\",\n",
    "                                                                                            \"TNM\",\n",
    "                                                                                            \"SMOKER\",\n",
    "                                                                                            \"CANCER_TYPE_DETAILED\"],\n",
    "                                   skiprows = 5)\n",
    "esoph2_patient_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SAMPLE_ID</th>\n",
       "      <th>PATIENT_ID</th>\n",
       "      <th>SEX</th>\n",
       "      <th>TUMOR_STAGE</th>\n",
       "      <th>AGE</th>\n",
       "      <th>NODAL_STAGE</th>\n",
       "      <th>NUMBER_OF_LYMPHNODES_POSITIVE</th>\n",
       "      <th>SMOKER</th>\n",
       "      <th>TOBACCO_SMOKING_HISTORY_INDICATOR</th>\n",
       "      <th>OS_STATUS</th>\n",
       "      <th>OS_MONTHS</th>\n",
       "      <th>DFS_STATUS</th>\n",
       "      <th>DFS_MONTHS</th>\n",
       "      <th>CANCER_TYPE_DETAILED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OSCJM-PT35-37-T</td>\n",
       "      <td>OSCJM-PT35-37</td>\n",
       "      <td>Male</td>\n",
       "      <td>T4</td>\n",
       "      <td>70</td>\n",
       "      <td>N2b</td>\n",
       "      <td>7</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>6.9</td>\n",
       "      <td>Recurred/Progressed</td>\n",
       "      <td>6.3</td>\n",
       "      <td>Head and Neck Squamous Cell Carcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OSCJM-PT07-43-T</td>\n",
       "      <td>OSCJM-PT07-43</td>\n",
       "      <td>Male</td>\n",
       "      <td>T4</td>\n",
       "      <td>50</td>\n",
       "      <td>N2b</td>\n",
       "      <td>2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>51.1</td>\n",
       "      <td>Diseasefree</td>\n",
       "      <td>48.6</td>\n",
       "      <td>Head and Neck Squamous Cell Carcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OSCJM-PT05-91-T</td>\n",
       "      <td>OSCJM-PT05-91</td>\n",
       "      <td>Male</td>\n",
       "      <td>T3</td>\n",
       "      <td>41</td>\n",
       "      <td>N1</td>\n",
       "      <td>1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>110.8</td>\n",
       "      <td>Diseasefree</td>\n",
       "      <td>1.4</td>\n",
       "      <td>Head and Neck Squamous Cell Carcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OSCJM-PTC-139-T</td>\n",
       "      <td>OSCJM-PTC-139</td>\n",
       "      <td>Male</td>\n",
       "      <td>T3</td>\n",
       "      <td>78</td>\n",
       "      <td>N1</td>\n",
       "      <td>1</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>10.8</td>\n",
       "      <td>Recurred/Progressed</td>\n",
       "      <td>9.7</td>\n",
       "      <td>Head and Neck Squamous Cell Carcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OSCJM-PT01-166-T</td>\n",
       "      <td>OSCJM-PT01-166</td>\n",
       "      <td>Male</td>\n",
       "      <td>T3</td>\n",
       "      <td>63</td>\n",
       "      <td>N0</td>\n",
       "      <td>0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>40.9</td>\n",
       "      <td>Diseasefree</td>\n",
       "      <td>39.0</td>\n",
       "      <td>Head and Neck Squamous Cell Carcinoma</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          SAMPLE_ID      PATIENT_ID   SEX TUMOR_STAGE  AGE NODAL_STAGE  \\\n",
       "0   OSCJM-PT35-37-T   OSCJM-PT35-37  Male          T4   70         N2b   \n",
       "1   OSCJM-PT07-43-T   OSCJM-PT07-43  Male          T4   50         N2b   \n",
       "2   OSCJM-PT05-91-T   OSCJM-PT05-91  Male          T3   41          N1   \n",
       "3   OSCJM-PTC-139-T   OSCJM-PTC-139  Male          T3   78          N1   \n",
       "4  OSCJM-PT01-166-T  OSCJM-PT01-166  Male          T3   63          N0   \n",
       "\n",
       "   NUMBER_OF_LYMPHNODES_POSITIVE SMOKER TOBACCO_SMOKING_HISTORY_INDICATOR  \\\n",
       "0                              7    Yes                               Yes   \n",
       "1                              2    Yes                               Yes   \n",
       "2                              1    Yes                               Yes   \n",
       "3                              1     No                                No   \n",
       "4                              0    Yes                               Yes   \n",
       "\n",
       "  OS_STATUS  OS_MONTHS           DFS_STATUS  DFS_MONTHS  \\\n",
       "0  DECEASED        6.9  Recurred/Progressed         6.3   \n",
       "1    LIVING       51.1          Diseasefree        48.6   \n",
       "2    LIVING      110.8          Diseasefree         1.4   \n",
       "3  DECEASED       10.8  Recurred/Progressed         9.7   \n",
       "4    LIVING       40.9          Diseasefree        39.0   \n",
       "\n",
       "                    CANCER_TYPE_DETAILED  \n",
       "0  Head and Neck Squamous Cell Carcinoma  \n",
       "1  Head and Neck Squamous Cell Carcinoma  \n",
       "2  Head and Neck Squamous Cell Carcinoma  \n",
       "3  Head and Neck Squamous Cell Carcinoma  \n",
       "4  Head and Neck Squamous Cell Carcinoma  "
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hn3_patient_df = pd.read_table(\"../hn3_data_clinical.txt\", usecols = lambda x: x in    [\"SAMPLE_ID\",\n",
    "                                                                                        \"PATIENT_ID\", \n",
    "                                                                                        \"SEX\",\n",
    "                                                                                        \"TUMOR_STAGE\",\n",
    "                                                                                        \"AGE\",\n",
    "                                                                                        \"NODAL_STAGE\",\n",
    "                                                                                        \"NUMBER_OF_LYMPHNODES_POSITIVE\",\n",
    "                                                                                        \"SMOKER\",\n",
    "                                                                                        \"TOBACCO_SMOKING_HISTORY_INDICATOR\",\n",
    "                                                                                        \"OS_STATUS\",\n",
    "                                                                                        \"OS_MONTHS\", \n",
    "                                                                                        \"DFS_STATUS\",\n",
    "                                                                                        \"DFS_MONTHS\",\n",
    "                                                                                        \"CANCER_TYPE_DETAILED\"],\n",
    "                                   skiprows = 5)\n",
    "hn3_patient_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SAMPLE_ID</th>\n",
       "      <th>PATIENT_ID</th>\n",
       "      <th>OS_STATUS</th>\n",
       "      <th>OS_MONTHS</th>\n",
       "      <th>CANCER_TYPE_DETAILED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-BA-4074-01</td>\n",
       "      <td>TCGA-BA-4074-01</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>15.15</td>\n",
       "      <td>Head and Neck Squamous Cell Carcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-BA-4076-01</td>\n",
       "      <td>TCGA-BA-4076-01</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>13.63</td>\n",
       "      <td>Head and Neck Squamous Cell Carcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-BA-4078-01</td>\n",
       "      <td>TCGA-BA-4078-01</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>9.07</td>\n",
       "      <td>Head and Neck Squamous Cell Carcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-BA-5149-01</td>\n",
       "      <td>TCGA-BA-5149-01</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>8.15</td>\n",
       "      <td>Head and Neck Squamous Cell Carcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-BA-5151-01</td>\n",
       "      <td>TCGA-BA-5151-01</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>6.24</td>\n",
       "      <td>Head and Neck Squamous Cell Carcinoma</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         SAMPLE_ID       PATIENT_ID OS_STATUS  OS_MONTHS  \\\n",
       "0  TCGA-BA-4074-01  TCGA-BA-4074-01  DECEASED      15.15   \n",
       "1  TCGA-BA-4076-01  TCGA-BA-4076-01  DECEASED      13.63   \n",
       "2  TCGA-BA-4078-01  TCGA-BA-4078-01  DECEASED       9.07   \n",
       "3  TCGA-BA-5149-01  TCGA-BA-5149-01    LIVING       8.15   \n",
       "4  TCGA-BA-5151-01  TCGA-BA-5151-01    LIVING       6.24   \n",
       "\n",
       "                    CANCER_TYPE_DETAILED  \n",
       "0  Head and Neck Squamous Cell Carcinoma  \n",
       "1  Head and Neck Squamous Cell Carcinoma  \n",
       "2  Head and Neck Squamous Cell Carcinoma  \n",
       "3  Head and Neck Squamous Cell Carcinoma  \n",
       "4  Head and Neck Squamous Cell Carcinoma  "
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hn4_patient_df = pd.read_table(\"../hn4_data_clinical.txt\", usecols = lambda x: x in    [\"SAMPLE_ID\",\n",
    "                                                                                        \"PATIENT_ID\",\n",
    "                                                                                        \"OS_STATUS\",\n",
    "                                                                                        \"OS_MONTHS\", \n",
    "                                                                                        \"CANCER_TYPE_DETAILED\"],\n",
    "                                   skiprows = 5)\n",
    "hn4_patient_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PATIENT_ID</th>\n",
       "      <th>HISTOLOGICAL_DIAGNOSIS</th>\n",
       "      <th>GENDER</th>\n",
       "      <th>LYMPH_NODES_EXAMINED_HE_COUNT</th>\n",
       "      <th>AJCC_TUMOR_PATHOLOGIC_PT</th>\n",
       "      <th>AJCC_NODES_PATHOLOGIC_PN</th>\n",
       "      <th>AJCC_METASTASIS_PATHOLOGIC_PM</th>\n",
       "      <th>AJCC_PATHOLOGIC_TUMOR_STAGE</th>\n",
       "      <th>GRADE</th>\n",
       "      <th>TOBACCO_SMOKING_HISTORY_INDICATOR</th>\n",
       "      <th>...</th>\n",
       "      <th>AGE</th>\n",
       "      <th>CLIN_M_STAGE</th>\n",
       "      <th>CLIN_N_STAGE</th>\n",
       "      <th>CLIN_T_STAGE</th>\n",
       "      <th>CLINICAL_STAGE</th>\n",
       "      <th>TUMOR_TISSUE_SITE</th>\n",
       "      <th>OS_STATUS</th>\n",
       "      <th>OS_MONTHS</th>\n",
       "      <th>DFS_STATUS</th>\n",
       "      <th>DFS_MONTHS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-BA-4074</td>\n",
       "      <td>Head &amp; Neck Squamous Cell Carcinoma</td>\n",
       "      <td>MALE</td>\n",
       "      <td>5</td>\n",
       "      <td>T2</td>\n",
       "      <td>N2c</td>\n",
       "      <td>M0</td>\n",
       "      <td>Stage IVA</td>\n",
       "      <td>G3</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>69</td>\n",
       "      <td>M0</td>\n",
       "      <td>N2c</td>\n",
       "      <td>T3</td>\n",
       "      <td>Stage IVA</td>\n",
       "      <td>Head and Neck</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>15.18</td>\n",
       "      <td>Recurred/Progressed</td>\n",
       "      <td>13.01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-BA-4075</td>\n",
       "      <td>Head &amp; Neck Squamous Cell Carcinoma</td>\n",
       "      <td>MALE</td>\n",
       "      <td>0</td>\n",
       "      <td>T3</td>\n",
       "      <td>N0</td>\n",
       "      <td>M0</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>G2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>49</td>\n",
       "      <td>M0</td>\n",
       "      <td>N1</td>\n",
       "      <td>T4a</td>\n",
       "      <td>Stage IVA</td>\n",
       "      <td>Head and Neck</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>9.30</td>\n",
       "      <td>Recurred/Progressed</td>\n",
       "      <td>7.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-BA-4076</td>\n",
       "      <td>Head &amp; Neck Squamous Cell Carcinoma</td>\n",
       "      <td>MALE</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>TX</td>\n",
       "      <td>NX</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>G2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>39</td>\n",
       "      <td>M0</td>\n",
       "      <td>N2c</td>\n",
       "      <td>T3</td>\n",
       "      <td>Stage IVA</td>\n",
       "      <td>Head and Neck</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>13.63</td>\n",
       "      <td>Recurred/Progressed</td>\n",
       "      <td>9.40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-BA-4077</td>\n",
       "      <td>Head &amp; Neck Squamous Cell Carcinoma</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>T4a</td>\n",
       "      <td>N0</td>\n",
       "      <td>M0</td>\n",
       "      <td>Stage IVA</td>\n",
       "      <td>G2</td>\n",
       "      <td>4</td>\n",
       "      <td>...</td>\n",
       "      <td>45</td>\n",
       "      <td>M0</td>\n",
       "      <td>N3</td>\n",
       "      <td>T4b</td>\n",
       "      <td>Stage IVB</td>\n",
       "      <td>Head and Neck</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>37.25</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>[Not Available]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-BA-4078</td>\n",
       "      <td>Head &amp; Neck Squamous Cell Carcinoma</td>\n",
       "      <td>MALE</td>\n",
       "      <td>4</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>G2</td>\n",
       "      <td>4</td>\n",
       "      <td>...</td>\n",
       "      <td>83</td>\n",
       "      <td>M0</td>\n",
       "      <td>N2a</td>\n",
       "      <td>T2</td>\n",
       "      <td>Stage IVA</td>\n",
       "      <td>Head and Neck</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>9.07</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>[Not Available]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     PATIENT_ID               HISTOLOGICAL_DIAGNOSIS  GENDER  \\\n",
       "0  TCGA-BA-4074  Head & Neck Squamous Cell Carcinoma    MALE   \n",
       "1  TCGA-BA-4075  Head & Neck Squamous Cell Carcinoma    MALE   \n",
       "2  TCGA-BA-4076  Head & Neck Squamous Cell Carcinoma    MALE   \n",
       "3  TCGA-BA-4077  Head & Neck Squamous Cell Carcinoma  FEMALE   \n",
       "4  TCGA-BA-4078  Head & Neck Squamous Cell Carcinoma    MALE   \n",
       "\n",
       "  LYMPH_NODES_EXAMINED_HE_COUNT AJCC_TUMOR_PATHOLOGIC_PT  \\\n",
       "0                             5                       T2   \n",
       "1                             0                       T3   \n",
       "2               [Not Available]                       TX   \n",
       "3               [Not Available]                      T4a   \n",
       "4                             4          [Not Available]   \n",
       "\n",
       "  AJCC_NODES_PATHOLOGIC_PN AJCC_METASTASIS_PATHOLOGIC_PM  \\\n",
       "0                      N2c                            M0   \n",
       "1                       N0                            M0   \n",
       "2                       NX               [Not Available]   \n",
       "3                       N0                            M0   \n",
       "4          [Not Available]               [Not Available]   \n",
       "\n",
       "  AJCC_PATHOLOGIC_TUMOR_STAGE GRADE TOBACCO_SMOKING_HISTORY_INDICATOR  \\\n",
       "0                   Stage IVA    G3                                 2   \n",
       "1                   Stage III    G2                                 2   \n",
       "2             [Not Available]    G2                                 2   \n",
       "3                   Stage IVA    G2                                 4   \n",
       "4             [Not Available]    G2                                 4   \n",
       "\n",
       "        ...        AGE CLIN_M_STAGE CLIN_N_STAGE CLIN_T_STAGE CLINICAL_STAGE  \\\n",
       "0       ...         69           M0          N2c           T3      Stage IVA   \n",
       "1       ...         49           M0           N1          T4a      Stage IVA   \n",
       "2       ...         39           M0          N2c           T3      Stage IVA   \n",
       "3       ...         45           M0           N3          T4b      Stage IVB   \n",
       "4       ...         83           M0          N2a           T2      Stage IVA   \n",
       "\n",
       "  TUMOR_TISSUE_SITE OS_STATUS OS_MONTHS           DFS_STATUS       DFS_MONTHS  \n",
       "0     Head and Neck  DECEASED     15.18  Recurred/Progressed            13.01  \n",
       "1     Head and Neck  DECEASED      9.30  Recurred/Progressed             7.75  \n",
       "2     Head and Neck  DECEASED     13.63  Recurred/Progressed             9.40  \n",
       "3     Head and Neck  DECEASED     37.25      [Not Available]  [Not Available]  \n",
       "4     Head and Neck  DECEASED      9.07      [Not Available]  [Not Available]  \n",
       "\n",
       "[5 rows x 23 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hn5_patient_df = pd.read_table(\"../hn5_data_bcr_clinical_data_patient.txt\", usecols = lambda x: x in [\"PATIENT_ID\",  \n",
    "                                                                                                      \"HISTOLOGICAL_DIAGNOSIS\",\n",
    "                                                                                                      \"GENDER\",\n",
    "                                                                                                      \"LYMPH_NODES_EXAMINED_HE_COUNT\",\n",
    "                                                                                                      \"AJCC_TUMOR_PATHOLOGIC_PT\",\n",
    "                                                                                                      \"AJCC_NODES_PATHOLOGIC_PN\",\n",
    "                                                                                                      \"AJCC_METASTASIS_PATHOLOGIC_PM\",\n",
    "                                                                                                      \"AJCC_PATHOLOGIC_TUMOR_STAGE\",\n",
    "                                                                                                      \"GRADE\",\n",
    "                                                                                                      \"TOBACCO_SMOKING_HISTORY_INDICATOR\", \n",
    "                                                                                                      \"SMOKING_YEAR_STARTED\", \n",
    "                                                                                                      \"SMOKING_YEAR_STOPPED\",\n",
    "                                                                                                      \"SMOKING_PACK_YEARS\",\n",
    "                                                                                                      \"AGE\",\n",
    "                                                                                                      \"CLIN_M_STAGE\",\n",
    "                                                                                                      \"CLIN_N_STAGE\",\n",
    "                                                                                                      \"CLIN_T_STAGE\",\n",
    "                                                                                                      \"CLINICAL_STAGE\", \n",
    "                                                                                                      \"TUMOR_TISSUE_SITE\",\n",
    "                                                                                                      \"OS_STATUS\",\n",
    "                                                                                                      \"OS_MONTHS\",\n",
    "                                                                                                      \"DFS_STATUS\",\n",
    "                                                                                                      \"DFS_MONTHS\"],\n",
    "                            skiprows = 4)\n",
    "hn5_patient_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PATIENT_ID</th>\n",
       "      <th>HISTOLOGICAL_DIAGNOSIS</th>\n",
       "      <th>GENDER</th>\n",
       "      <th>INITIAL_PATHOLOGIC_DX_YEAR</th>\n",
       "      <th>AJCC_TUMOR_PATHOLOGIC_PT</th>\n",
       "      <th>AJCC_NODES_PATHOLOGIC_PN</th>\n",
       "      <th>AJCC_METASTASIS_PATHOLOGIC_PM</th>\n",
       "      <th>AJCC_PATHOLOGIC_TUMOR_STAGE</th>\n",
       "      <th>TOBACCO_SMOKING_HISTORY_INDICATOR</th>\n",
       "      <th>SMOKING_YEAR_STARTED</th>\n",
       "      <th>SMOKING_YEAR_STOPPED</th>\n",
       "      <th>SMOKING_PACK_YEARS</th>\n",
       "      <th>AGE</th>\n",
       "      <th>TUMOR_TISSUE_SITE</th>\n",
       "      <th>OS_STATUS</th>\n",
       "      <th>OS_MONTHS</th>\n",
       "      <th>DFS_STATUS</th>\n",
       "      <th>DFS_MONTHS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-18-3407</td>\n",
       "      <td>Lung Squamous Cell Carcinoma</td>\n",
       "      <td>MALE</td>\n",
       "      <td>2003</td>\n",
       "      <td>T2</td>\n",
       "      <td>N0</td>\n",
       "      <td>M0</td>\n",
       "      <td>Stage IB</td>\n",
       "      <td>3</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>1988</td>\n",
       "      <td>40</td>\n",
       "      <td>72</td>\n",
       "      <td>Lung</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>4.47</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>[Not Available]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-18-3408</td>\n",
       "      <td>Lung Squamous Cell Carcinoma</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>2004</td>\n",
       "      <td>T2</td>\n",
       "      <td>N0</td>\n",
       "      <td>M0</td>\n",
       "      <td>Stage IB</td>\n",
       "      <td>4</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>2004</td>\n",
       "      <td>30</td>\n",
       "      <td>77</td>\n",
       "      <td>Lung</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>75.69</td>\n",
       "      <td>Recurred/Progressed</td>\n",
       "      <td>58.90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-18-3409</td>\n",
       "      <td>Lung Squamous Cell Carcinoma</td>\n",
       "      <td>MALE</td>\n",
       "      <td>2004</td>\n",
       "      <td>T1</td>\n",
       "      <td>N0</td>\n",
       "      <td>M0</td>\n",
       "      <td>Stage IA</td>\n",
       "      <td>3</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>1974</td>\n",
       "      <td>20</td>\n",
       "      <td>74</td>\n",
       "      <td>Lung</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>123.09</td>\n",
       "      <td>Recurred/Progressed</td>\n",
       "      <td>75.26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-18-3410</td>\n",
       "      <td>Lung Squamous Cell Carcinoma</td>\n",
       "      <td>MALE</td>\n",
       "      <td>2004</td>\n",
       "      <td>T3</td>\n",
       "      <td>N0</td>\n",
       "      <td>M0</td>\n",
       "      <td>Stage IIB</td>\n",
       "      <td>3</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>81</td>\n",
       "      <td>Lung</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>4.80</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>[Not Available]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-18-3411</td>\n",
       "      <td>Lung Squamous Cell Carcinoma</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>2005</td>\n",
       "      <td>T2</td>\n",
       "      <td>N2</td>\n",
       "      <td>M0</td>\n",
       "      <td>Stage IIIA</td>\n",
       "      <td>2</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>[Not Available]</td>\n",
       "      <td>50</td>\n",
       "      <td>63</td>\n",
       "      <td>Lung</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>117.48</td>\n",
       "      <td>DiseaseFree</td>\n",
       "      <td>117.48</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     PATIENT_ID        HISTOLOGICAL_DIAGNOSIS  GENDER  \\\n",
       "0  TCGA-18-3407  Lung Squamous Cell Carcinoma    MALE   \n",
       "1  TCGA-18-3408  Lung Squamous Cell Carcinoma  FEMALE   \n",
       "2  TCGA-18-3409  Lung Squamous Cell Carcinoma    MALE   \n",
       "3  TCGA-18-3410  Lung Squamous Cell Carcinoma    MALE   \n",
       "4  TCGA-18-3411  Lung Squamous Cell Carcinoma  FEMALE   \n",
       "\n",
       "  INITIAL_PATHOLOGIC_DX_YEAR AJCC_TUMOR_PATHOLOGIC_PT  \\\n",
       "0                       2003                       T2   \n",
       "1                       2004                       T2   \n",
       "2                       2004                       T1   \n",
       "3                       2004                       T3   \n",
       "4                       2005                       T2   \n",
       "\n",
       "  AJCC_NODES_PATHOLOGIC_PN AJCC_METASTASIS_PATHOLOGIC_PM  \\\n",
       "0                       N0                            M0   \n",
       "1                       N0                            M0   \n",
       "2                       N0                            M0   \n",
       "3                       N0                            M0   \n",
       "4                       N2                            M0   \n",
       "\n",
       "  AJCC_PATHOLOGIC_TUMOR_STAGE TOBACCO_SMOKING_HISTORY_INDICATOR  \\\n",
       "0                    Stage IB                                 3   \n",
       "1                    Stage IB                                 4   \n",
       "2                    Stage IA                                 3   \n",
       "3                   Stage IIB                                 3   \n",
       "4                  Stage IIIA                                 2   \n",
       "\n",
       "  SMOKING_YEAR_STARTED SMOKING_YEAR_STOPPED SMOKING_PACK_YEARS AGE  \\\n",
       "0      [Not Available]                 1988                 40  72   \n",
       "1      [Not Available]                 2004                 30  77   \n",
       "2      [Not Available]                 1974                 20  74   \n",
       "3      [Not Available]      [Not Available]    [Not Available]  81   \n",
       "4      [Not Available]      [Not Available]                 50  63   \n",
       "\n",
       "  TUMOR_TISSUE_SITE OS_STATUS OS_MONTHS           DFS_STATUS       DFS_MONTHS  \n",
       "0              Lung  DECEASED      4.47      [Not Available]  [Not Available]  \n",
       "1              Lung  DECEASED     75.69  Recurred/Progressed            58.90  \n",
       "2              Lung    LIVING    123.09  Recurred/Progressed            75.26  \n",
       "3              Lung  DECEASED      4.80      [Not Available]  [Not Available]  \n",
       "4              Lung    LIVING    117.48          DiseaseFree           117.48  "
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lung2_patient_df = pd.read_table(\"../lung2_data_bcr_clinical_data_patient.txt\", usecols = lambda x: x in [\"PATIENT_ID\",  \n",
    "                                                                                                          \"HISTOLOGICAL_DIAGNOSIS\",\n",
    "                                                                                                          \"GENDER\",\n",
    "                                                                                                          \"INITIAL_PATHOLOGIC_DX_YEAR\",\n",
    "                                                                                                          \"AJCC_TUMOR_PATHOLOGIC_PT\",\n",
    "                                                                                                          \"AJCC_NODES_PATHOLOGIC_PN\",\n",
    "                                                                                                          \"AJCC_METASTASIS_PATHOLOGIC_PM\",\n",
    "                                                                                                          \"AJCC_PATHOLOGIC_TUMOR_STAGE\",\n",
    "                                                                                                          \"TOBACCO_SMOKING_HISTORY_INDICATOR\", \n",
    "                                                                                                          \"SMOKING_YEAR_STARTED\", \n",
    "                                                                                                          \"SMOKING_YEAR_STOPPED\",\n",
    "                                                                                                          \"SMOKING_PACK_YEARS\",\n",
    "                                                                                                          \"GRADE\",\n",
    "                                                                                                          \"AGE\",\n",
    "                                                                                                          \"TUMOR_TISSUE_SITE\",\n",
    "                                                                                                          \"OS_STATUS\",\n",
    "                                                                                                          \"OS_MONTHS\",\n",
    "                                                                                                          \"DFS_STATUS\",\n",
    "                                                                                                          \"DFS_MONTHS\"],\n",
    "                            skiprows = 4)\n",
    "lung2_patient_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SAMPLE_ID</th>\n",
       "      <th>PATIENT_ID</th>\n",
       "      <th>AGE</th>\n",
       "      <th>GENDER</th>\n",
       "      <th>SMOKER</th>\n",
       "      <th>DFS_STATUS</th>\n",
       "      <th>OS_STATUS</th>\n",
       "      <th>CANCER_TYPE_DETAILED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>S00-28455-TP-NT</td>\n",
       "      <td>S00-28455</td>\n",
       "      <td>73</td>\n",
       "      <td>Female</td>\n",
       "      <td>No</td>\n",
       "      <td>Recurred/Progressed</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>Cutaneous Squamous Cell Carcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>S00-35182-TP-NT</td>\n",
       "      <td>S00-35182</td>\n",
       "      <td>84</td>\n",
       "      <td>Male</td>\n",
       "      <td>Yes</td>\n",
       "      <td>DiseaseFree</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>Cutaneous Squamous Cell Carcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>S01-27743-TP-NT</td>\n",
       "      <td>S01-27743</td>\n",
       "      <td>72</td>\n",
       "      <td>Female</td>\n",
       "      <td>No</td>\n",
       "      <td>Recurred/Progressed</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>Cutaneous Squamous Cell Carcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>S02-14875</td>\n",
       "      <td>S02-14785</td>\n",
       "      <td>48</td>\n",
       "      <td>Male</td>\n",
       "      <td>No</td>\n",
       "      <td>DiseaseFree</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>Cutaneous Squamous Cell Carcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>S02-15015-TP-S00-35182-NT</td>\n",
       "      <td>S02-15015</td>\n",
       "      <td>75</td>\n",
       "      <td>Male</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Recurred/Progressed</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>Cutaneous Squamous Cell Carcinoma</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   SAMPLE_ID PATIENT_ID  AGE  GENDER SMOKER  \\\n",
       "0            S00-28455-TP-NT  S00-28455   73  Female     No   \n",
       "1            S00-35182-TP-NT  S00-35182   84    Male    Yes   \n",
       "2            S01-27743-TP-NT  S01-27743   72  Female     No   \n",
       "3                  S02-14875  S02-14785   48    Male     No   \n",
       "4  S02-15015-TP-S00-35182-NT  S02-15015   75    Male    Yes   \n",
       "\n",
       "            DFS_STATUS OS_STATUS               CANCER_TYPE_DETAILED  \n",
       "0  Recurred/Progressed    LIVING  Cutaneous Squamous Cell Carcinoma  \n",
       "1          DiseaseFree  DECEASED  Cutaneous Squamous Cell Carcinoma  \n",
       "2  Recurred/Progressed    LIVING  Cutaneous Squamous Cell Carcinoma  \n",
       "3          DiseaseFree    LIVING  Cutaneous Squamous Cell Carcinoma  \n",
       "4  Recurred/Progressed    LIVING  Cutaneous Squamous Cell Carcinoma  "
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "skin_patient_df = pd.read_table(\"../skin_patient_data_clinical.txt\", usecols = lambda x: x in  [\"SAMPLE_ID\",\n",
    "                                                                                        \"PATIENT_ID\",\n",
    "                                                                                        \"AGE\",\n",
    "                                                                                        \"GENDER\", \n",
    "                                                                                        \"SMOKER\",\n",
    "                                                                                        \"DFS_STATUS\",\n",
    "                                                                                        \"OS_STATUS\",\n",
    "                                                                                        \"PFS_MONTHS\"\n",
    "                                                                                        \"OS_MONTHS\",\n",
    "                                                                                        \"CANCER_TYPE_DETAILED\"],\n",
    "                                   skiprows = 5)\n",
    "skin_patient_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Prepping the data for the machine learning pipeline\n",
    "\n",
    "The previous data analysis done in R was done using mutation data that was flattened across the study. That is, for each study, a column was provided in the dataframe for each of the genes mutated in all of the datasets and a value of 1, if the mutation was identified in the study, or 0, if not, was assigned to that gene for that study. However, to employ machine learning, I need to flatten over each patient in each study since each patient has a different clinical outcome, which is what I am trying to predict. \n",
    "\n",
    "The following code flattens each genomic dataframe as follows:\n",
    "\n",
    "patient_ID (this needs to be extracted from barcode ID) | gene1_impact(0 for not mutated, 1 for moderate, and 2 for high) | gene2_impact | ...\n",
    "\n",
    "Thus, the mutation status of each gene is a feature in the matrix used for machine learning."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#first extract patient ID from barcode id \n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
